qiskit/transpiler/passes/routing/utils.py

```diff
@@ -0,0 +1,211 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utility functions for routing"""
+
+import numpy as np
+from qiskit.transpiler.layout import Layout
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.circuit import QuantumRegister
+
+
+def route_cf_multiblock(tpass, cf_opnode, current_layout, qregs, root_dag, seed=None):
+    """Transpile control flow instructions which may contain multiple
+    blocks (e.g. IfElseOp). Since each control flow block may yield a
+    different layout, this function applies swaps to the shorter depth
+    blocks to make all final layouts match.
+
+    Args:
+        tpass (BasePass): Transpiler pass object to use recursively.
+        cf_opnode (DAGOpNode): multiblock instruction node e.g. IfElseOp.
+        current_layout (Layout): The current layout at the start of the instruction.
+        qregs (list(QuantumRegister)): quantum registers for circuit
+        root_dag (DAGCircuit): root dag of compilation
+        seed (int): seed for RNG of internal layout transformation.
+    Returns:
+        IfElseOp: transpiled control flow operation
+        final_layout (Layout): layout after instruction
```

Comment:
This return information is incomplete now, and the fact that we even need to return 3 things from this function suggests that it might not be the cleanest API boundary.